dist/scripts/services/cookie.js

```diff
@@ -0,0 +1,17 @@
+(function() {
+  function BlocChatCookies($cookies, $uibModal) {
+    var currentUser = $cookies.get('blocChatCurrentUser');
+    if (!currentUser || currentUser === '') {
+        $uibModal.open ({
```

Comment:
You could store a reference to the modal here like `var modalInstance = $uibModal.open({})` and then use the callback to store the username. (look below)